### DISCUSSION TITLE
Use of milestones

### DISCUSSION THREAD

**Discussion by stephan-herrmann** [eclipse-jdt/eclipse.jdt.core#2627] [General]
Today, for the first time since the move to github, a bug reporter asked me which "milestone" contains a given fix.

I readily admit that I forgot to mark it on the issue.

So, how to we handle milestones?

* Should all merged fixes have the milestone set?
* Should it be set on the issue or on the PR?
* Are milestones used for some standard queries like bugs fixed per release?
* Is there any help to ensure that this setting is never forgotten?

**Comment by iloveeclipse** [comment]
> * Should all merged fixes have the milestone set?

IMO yes, as this greatly simplifies my life be looking in which release the regression was introduced or from which release some bug should be fixed.
 
> * Should it be set on the issue or on the PR?

I always do that, but I've given up asking people to use proper commit messages (including both reasoning and link to issue), if I would start asking them for using milestones I would be considered fully mad.

> * Are milestones used for some standard queries like bugs fixed per release?

See https://github.com/eclipse-jdt/eclipse.jdt.core/milestone/43?closed=1

> * Is there any help to ensure that this setting is never forgotten?

No idea

**Reply by akurtakov** [reply]
Theia project automated setting milestone on PR merge via https://github.com/EstherPerelman/theia/commit/d00dca2480be05d0e052a603c9950dde49ed4ae8 . With some adjustment this should be working for JDT too(I would happily copy to rest of the projects in SDK if/when someone spends the time to make it work).

**Reply by iloveeclipse** [reply]
> I would happily copy to rest of the projects in SDK if/when someone spends the time to make it work).

Not sure what do you mean by that, which projects and which work?

**Reply by akurtakov** [reply]
If someone makes auto assigning milestone on merge work in e.g. JDT(core) by adopting Theia's script (or some other similar), I'll copy the approach to Platform/PDE/Equinox.

**Reply by iloveeclipse** [reply]
I've created https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/2203.
Let start with the platform first and provide common solution for all :-)

**Comment by jukzi** [comment]
closing . to be continued on aggregator
